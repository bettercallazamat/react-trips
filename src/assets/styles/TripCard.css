.TripCard {
  display: flex;
  align-items: center;
  margin: 30px 20px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  transition: 0.3s;
  height: 100%;
}

.TripCard .trip-img {
  width: 30%;
}

.TripCard img {
  height: 230px;
  width: 100%;
  object-fit: cover;
}

.TripCard .info {
  padding: 20px;
  width: 50%;
}

.TripCard .title {
  color: #ef4b27;
  font-size: 20px;
  margin-bottom: 10px;
}

.TripCard .desc {
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  overflow: hidden;
  line-height: 25px;
}

.TripCard .extra-info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 20%;
  height: 230px;
  text-align: center;
  border-left: 1px solid #888;
}

.TripCard .trip-cost {
  font-size: 28px;
  margin-bottom: 6px;
  letter-spacing: 1.5px;
}

.TripCard .trip-cost-comment {
  font-size: 14px;
  margin-bottom: 6px;
}

.TripCard .details-link {
  width: fit-content;
  margin-top: 6px;
  background-color: #e95939;
  padding: 12px 24px;
  text-decoration: none;
  border-radius: 3px;
  color: #fff;
  font-weight: bold;
}
