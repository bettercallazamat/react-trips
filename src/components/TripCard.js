import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../assets/styles/TripCard.css';

const TripCard = ({ trip }) => (
  <div className="TripCard">
    <Link to={`/trip/${trip.id}`} key={trip.id} className="trip-img">
      <img alt="trip" src={trip.imageUrl} />
    </Link>
    <div className="info">
      <h3 className="title">{trip.title}</h3>
      <p className="desc">{trip.description}</p>
    </div>
    <div className="extra-info">
      <p className="trip-cost">{trip.cost}</p>
      <p className="trip-cost-comment">* per person</p>
      <Link to={`/trip/${trip.id}`} key={trip.id} className="details-link">
        <p>DETAILS</p>
      </Link>
    </div>
  </div>
);

TripCard.propTypes = {
  trip: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    description: PropTypes.string,
    cost: PropTypes.string,
    imageUrl: PropTypes.string,
  }).isRequired,
};

export default TripCard;
