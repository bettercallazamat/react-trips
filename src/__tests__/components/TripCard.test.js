import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import TripCard from '../../components/TripCard';

describe('TripCard', () => {
  configure({ adapter: new Adapter() })
  const trip = {
    id: 1,
    title: "ALAMEDIN GORGE",
    description: "The Alamedin Gorge is one of the four most visited and popular gorges of the Kirghiz Range.",
    imageUrl: "http://example.com/image/1",
    cost: "500 KGS",
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<Router><TripCard trip={trip} /></Router>);
    expect(wrapper).toMatchSnapshot();
  });
});