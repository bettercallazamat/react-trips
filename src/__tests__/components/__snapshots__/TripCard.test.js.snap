// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripCard should match the snapshot 1`] = `ShallowWrapper {}`;
